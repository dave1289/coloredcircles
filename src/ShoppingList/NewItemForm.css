.NewItemForm-button {
   margin-left: 10px;
}

.NewItemForm-name {
   margin: 20px 30px 0 30px;
}

.NewItemForm-qty {
   margin: 20px 30px 0 30px;
}
