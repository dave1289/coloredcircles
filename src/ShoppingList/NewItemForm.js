import React, { useState } from "react";
import "./NewItemForm.css"

const NewItemForm = ({ addItem }) => {
   const INITIAL_STATE = {
      name: "",
      qty: "",
   }
   const [formData, setFormData] = useState(INITIAL_STATE)
   const handleChange = e => {
      const { name, value } = e.target
      setFormData(data => ({
         ...data,
         [name]: value
      }))
   }
   const handleSubmit = (e) => {
      e.preventDefault();
      addItem({...formData})
      setFormData(INITIAL_STATE)
   }
   return (
      <form className="NewItemForm text-center" onSubmit={handleSubmit}>
         <label htmlFor="name">Product Name:</label>
         <input
            placeholder="product name"
            className="NewItemForm-name"
            name="name"
            id="name"
            value={formData.name}
            onChange={handleChange}></input>
         <label htmlFor="qty">Quantity: {formData.qty} </label>
         <input
            className="NewItemForm-qty"
            type="range" min="1" max="10"
            id="qty"
            name="qty"
            onChange={handleChange}
            value={formData.qty}
            />
         <button className="NewItemForm-button btn btn-primary">Add Item</button>
      </form>
   )
}


export default NewItemForm