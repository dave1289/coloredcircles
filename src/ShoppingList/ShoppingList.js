import React, { useState } from "react";
import Item from "./Item";
import NewItemForm from "./NewItemForm";
import { v4 as uuid } from 'uuid';

const ShoppingList = () => {
   const INITIAL_STATE = [
      { id: 1, name: 'Peanut Butter', qty: 2 },
      { id: 2, name: 'Milk', qty: 1 }
   ]
   const addItem = (newItem) => {
      setItems(items => [...items, {...newItem, id: uuid()}])
   }
   const [items, setItems] = useState(INITIAL_STATE)
   return (
      <div>
         <h3 className="display-4">Shopping List</h3>
         <div>
            {items.map(({ id, name, qty }) => <Item name={name} qty={qty} key={id} />)}
         </div>
         <NewItemForm addItem={addItem} />
      </div>
   )
}

export default ShoppingList