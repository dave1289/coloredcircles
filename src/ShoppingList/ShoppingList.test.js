import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ShoppingList from "./ShoppingList";

it("renders without crashing", () => {
   render(<ShoppingList />);
})

it("matches snapshot", () => {
   const { asFragment } = render(<ShoppingList />);
   expect(asFragment()).toMatchSnapshot();
})

it("should add new item", () => {
   const {queryByText} = render(<ShoppingList />);
   expect(queryByText('Product Name: Chocolate Milk')).not.toBeInTheDocument();
})

it("should add new item", () => {
   const {queryByText, getByLabelText} = render(<ShoppingList />);
   const input = getByLabelText("Product Name:")
   const btn = queryByText("Add Item")
   fireEvent.change(input, {target: {value: 'Chocolate Milk'}})
   fireEvent.click(btn);
   expect(queryByText('Product Name: Chocolate Milk')).toBeInTheDocument();   
})