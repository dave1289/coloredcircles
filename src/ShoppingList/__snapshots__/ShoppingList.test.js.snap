// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="display-4"
    >
      Shopping List
    </h3>
    <div>
      <ul>
        <li>
          Product Name: Peanut Butter
        </li>
        <li>
          Quantity: 2
        </li>
      </ul>
      <ul>
        <li>
          Product Name: Milk
        </li>
        <li>
          Quantity: 1
        </li>
      </ul>
    </div>
    <form
      class="NewItemForm text-center"
    >
      <label
        for="name"
      >
        Product Name:
      </label>
      <input
        class="NewItemForm-name"
        id="name"
        name="name"
        placeholder="product name"
        value=""
      />
      <label
        for="qty"
      >
        Quantity:  
      </label>
      <input
        class="NewItemForm-qty"
        id="qty"
        max="10"
        min="1"
        name="qty"
        type="range"
        value=""
      />
      <button
        class="NewItemForm-button btn btn-primary"
      >
        Add Item
      </button>
    </form>
  </div>
</DocumentFragment>
`;
